portfolio/src/main/webapp/game.js

```diff
@@ -100,10 +117,106 @@ class Board {
       } else {
         this.board[row][col] = 2;
       }
+      this.numTiles++;
+      this.checkGameState(this.redTrue);
```

Comment:
Nit:
Since redTrue is a member variable of the class and only updated afterwards, it is cleaner to not pass it in at all and just look up the value in this.checkGameState.